packages/syft/src/syft/core/tensor/autodp/intermediate_gamma.py

```diff
@@ -258,3 +259,52 @@ def __mul__(self, other: Any) -> IntermediateGammaTensor:
             bias_tensor=bias_tensor,
             scalar_manager=self.scalar_manager,
         )
+
+    def __eq__(self, other: SupportedChainType) -> IntermediateGammaTensor:
+        if is_acceptable_simple_type(other):
+            term_data = (
+                self.term_tensor == other
+            )  # Need to check if shapes are broadcastable!!
+        elif isinstance(other, IntermediateGammaTensor):
+            # TODO: Check what actually needs to be identical to do an equality comparison
+            if (
+                self.child.shape == other.child.shape
+            ):  # also check if shapes are broadcastable
+                term_data = self.child == other.child
+            else:
+                raise Exception(
+                    f"Term Tensor shapes do not match for __eq__: {self.child} != len{other}"
+                )
+
+            if (
+                self.coeff_tensor.shape == other.coeff_tensor.shape
+            ):  # also check if shapes are broadcastable
+                coeff_data = self.coeff_tensor == other.coeff_tensor
+            else:
+                raise Exception(
+                    f"Coeff Tensor shapes do not match for __eq__: {self.child} != len{other}"
+                )
+
+            if (
+                self.bias_tensor.shape == other.bias_tensor.shape
+            ):  # also check if shapes are broadcastable
+                bias_data = self.bias_tensor == other.bias_tensor
+            else:
+                raise Exception(
+                    f"Bias Tensor shapes do not match for __eq__: {self.child} != len{other}"
+                )
+        elif isinstance(other, PassthroughTensor):
+            if (
+                self.child.shape == other.child.shape
+            ):  # also check if shapes are broadcastable
+                term_data = self.child == other.child
```

Comment:
Unfortunately I'm not sure that this can be the way in which we handle equality. This checks to see if the polynomial itself is equal but what we actually need to do is check whether the underlying result of the polynomial is equal. One could have two different polnyomials return the same value (they could both return 0, for example)

Reply:
(we can't do IntermediateGammaTensor until we modify with the clip at teh end of the polynomial as discussed at the last meeting I attended.)

Reply:
I fully agree, I haven't modified the code for IntermediateGammaTensors since before we had that meeting, and reduced the scope of this PR to just the methods for SingleEntity and RowEntity Phi Tensors

Reply:
Marking as resolved since there's another PR open for equality for IntermediateGammaTensors